### PR TITLE
groupcache: handler using mailgun's groupcache

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

**Comment by amery** [comment]
amended as I forgot the dummy `NewPool()`

**Comment by amery** [comment]
should I move the context helpers to `internal/`?

**Comment by amery** [comment]
WithSink moved to internal/ and NewHTTPPool/NewNoPeersPool added to groupcache @karasz

File: x/groupcache/groupcache.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+	return NewHTTPPoolOpts(self, nil)
+}
+
+// Pool implements a cache.Store using mailgun's groupcache.
+// Groupcache is global so the Pool object doesn't contain anything
+type Pool struct{}
+
+// NewNoPeersPool initialises groupcache with a PeerPicker that never find a peer
+// Due to groupcache's own limitations you have to choose to use either
+// HTTPPool or NoPeersPool. There can be only one.
```
Comment (karasz) [inline x/groupcache/groupcache.go:71]: "In the end, there can be only one" is the correct phrasing.

**Review by karasz — APPROVED** [review]
LGTM

File: x/groupcache/groupcache.go (hunk 1 above)
Comment (amery) [reply x/groupcache/groupcache.go:71]: Highlander?

**Comment by amery** [comment]
`v0.0.3` and `x/groupcache/v0.0.1` tagged

**Pull request merged**